pkg/tsdb/sqleng/sql_engine.go

```diff
@@ -150,7 +150,7 @@ func NewQueryDataHandler(config DataPluginConfiguration, queryResultTransformer
 	defer engineCache.Unlock()
 
 	if engine, present := engineCache.cache[config.DSInfo.ID]; present {
-		if updateTime := engineCache.updates[config.DSInfo.ID]; updateTime.Before(config.DSInfo.Updated) {
+		if updateTime := engineCache.updates[config.DSInfo.ID]; updateTime == config.DSInfo.Updated {
```

Comment:
Any reason `updateTime.Before` being used instead of equal?

Reply:
i think it is because when the update time in the cache is earlier than the current update time of configuration, we need to refresh the cache.

Reply:
is there any problem introduced here?

Reply:
Yes, if the cached `updateTime` is before the new `updateTime` it means you should create a new engine rather than return the existing one.

Reply:
then i think here we probably should put !updateTime.Before(config.DSInfo.Updated) since when the cache update is equal or later than the input config, we need to return the current cache.